mesheryctl/pkg/utils/helpers.go

```diff
@@ -693,3 +693,58 @@ func IsAdapterValid(manifestArr []Manifest, adapterManifest string) bool {
 
 	return false
 }
+
+// GetCurrentContext returns the current context name and context struct
+func GetCurrentContext(tempContext string) (string, config.Context, error) {
+	// if the user has mentioned a temporary context in the -c flag, change the context and proceed to reset
+	mctlCfg, err := config.GetMesheryCtl(viper.GetViper())
+	if err != nil {
+		return "", config.Context{}, errors.Wrap(err, "error processing config")
+	}
+
+	if tempContext == "" {
+		return mctlCfg.CurrentContext, mctlCfg.Contexts[mctlCfg.CurrentContext], nil
+	}
+	if tempCtxtStruct, val := mctlCfg.Contexts[tempContext]; val {
+		return tempContext, tempCtxtStruct, nil
+	}
+	// if the user specifies a context that is not in the config.yaml file, throw an error and show the available contexts
+	log.Errorf("\n\"%s\" context does not exist. The available contexts are:", tempContext)
+	for context := range mctlCfg.Contexts {
+		log.Errorf("%s", context)
+	}
+	return "", config.Context{}, errors.Errorf("\n\"%s\" context does not exist", tempContext)
+}
+
+// DownloadDockerComposeFile check if docker-compose.yaml exists, if not fetches the correct file based on current-context
+func DownloadDockerComposeFile(tempContext string, force bool) error {
+	if _, err := os.Stat(DockerComposeFile); os.IsNotExist(err) || force {
+		_, currCtx, err := GetCurrentContext(tempContext)
+		if err != nil {
+			return errors.Wrap(err, "failed to retrieve current-context")
```

Comment:
Why do we need to make this call again since we have already picked the current context or the temporary context before?

Reply:
Yes we did already get the context, but that is only in the case of `resetMesheryConfig`, it was done for logging purpose as mentioned in the other comment. For the rest of the areas where `DownloadDockerComposeFile` will be used, we will not find the `current-context` beforehand.

Reply:
In those cases too, can we use the same logic as we used here? One function which takes in the temporary context and returns either the temporary context or the current context. Another function which takes in the current/temporary context (whichever is set) and then downloads the corresponding version file. I think that approach would be neat and readable.

Adding @anirudhjain75 to this thread.

Reply:
If we are planning to add temporary-context to other commands, then I think you are correct. Currently, all the other commands except `resetMesheryConfig` download docker-compose file only if it is not found. Ideally, it should download the file if it does not exist or **if the docker-compose file does not match the current-context** (especially needed for temporary-context). The second part is not handled at the moment. Though making mesheryctl always download docker-compose file will be a pain for users not connected to the internet.

Reply:
The `--context` flag would be ideally a global flag that is available in all of the mesheryctl commands. This has yet to be implemented but it will be down the road. So we should be adding it in such a way that we won't have to rewrite it in the future.